include/picongpu/fields/MaxwellSolver/ArbitraryOrderFDTD/Weights.hpp

```diff
@@ -0,0 +1,85 @@
+/* Copyright 2013-2020 Klaus Steiniger, Sergei Bastrakov
+ *
+ * This file is part of PIConGPU.
+ *
+ * PIConGPU is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * PIConGPU is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with PIConGPU.
+ * If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+#include <cstdint>
+#include <cmath>
+#include "picongpu/plugins/radiation/utilities.hpp"
+
+
+namespace picongpu
+{
+namespace fields
+{
+namespace maxwellSolver
+{
+namespace aoFDTD
+{
+/** Compute weights of finite differences in
+ *
+ * @tparam T_neighbors Number of neighbors used to calculate
+ *                     the derivative from finite differences.
+ *                     Order of derivative approximation is
+ *                     2 * T_neighbors
+ */
+template< uint32_t T_neighbors >
+struct AOFDTDWeights
+{
+    HDINLINE constexpr AOFDTDWeights()
+    {
+        namespace powSpace = ::picongpu::plugins::radiation::util;
+        // Set initial value
+        weights[0] = 4.0_X * T_neighbors *
+            powSpace::pow( (
+                factorial(2 * T_neighbors)
+                / float_X( powSpace::pow(2.0_X, 2 * T_neighbors) * powSpace::pow( factorial(T_neighbors), 2) )
+            ), 2);
+
+        // Compute all other values
+        for ( uint32_t l = 1u; l < T_neighbors; ++l )
+        {
+            weights[l] = -1.0_X
```

Comment:
for `T_neighbors = 2` this would lead to:
```
w[0] = 9/8
w[1] = -9/8* 1/27 = -1  / (8*9) = -1/72
```

Reply:
ghrist_b0(m=2, l=0.5) =  1.125 ,  9/8 =  1.125
ghrist_b0(m=2, l=1.5) = -0.042, -1/72 = -0.0139

Reply:
Thanks - not intuitive to me, but definitely correct